Activities/Module13 Activity.md

```diff
@@ -146,3 +146,217 @@ In case your server is is not responding to requests after a long period of inac
+      name: FakeStackOverflow CI
+      on: # Controls when the action will run.
+      # Triggers the workflow on push or pull request events but only for the master branch. If you want to trigger the action on other branches, add here
+      push:
+         branches: [main]
+      pull_request:
+         branches: [main]
+
+      # Allows you to run this workflow manually from the Actions tab
+      workflow_dispatch:
+
+      # A workflow run is made up of one or more jobs that can run sequentially or in parallel
+      jobs:
+      build-and-test: #
+         # The type of runner that the job will run on
+         runs-on: ubuntu-20.04
+
+         # Steps represent a sequence of tasks that will be executed as part of the job
+         steps:
+            - name: Git Checkout
+            uses: actions/checkout@v4
+
+            - name: Set up Node.js 20.x
+            uses: actions/setup-node@v4
+            with:
+               node-version: "20.x"
+
+            - name: Launch MongoDB Server
+            uses: supercharge/mongodb-github-action@1.11.0
+            with:
+               mongodb-version: "7.0"
+
+            - name: Build backend service
+            if: ${{ always() }}
+            run: cd server; npm ci
+
+            - name: Test backend server
+            if: ${{ always() }}
+            env:
+               # Pass the environmental variables for the backend tests to use
+               MONGODB_URI: mongodb://localhost:27017
+            run: |
+               cd server
+
+               npm run start & sleep 10
+
+               echo "Checking if the server is running..."
+               curl --fail 'http://localhost:8000/question/getQuestion?order=newest&search=' || (echo "Server failed to start" && killall node && exit 1)
+
+               echo "Server started successfully. Now stopping..."
+               killall node
+
+               npm run test
+
+            - name: Lint backend
+            if: ${{ always() }}
+            run: cd server; npm run lint
+      ```
+
+## Step 4: Connect to GitHub for Continuous Deployment
+
+1. **Connect to GitHub Repository**
+   * In your Heroku app dashboard, scroll to the **Deployment method** section and select **GitHub**.
+   * Click **Connect to GitHub** and search for your repository.
+   * Select the repository containing your backend service code.
+
+   ![Connect to your Repo](Module13Assets/ConnectToGithub.png)
+
+2. **Configure Required Files**
+   * You need to make these changes to the package.json
+     * Move "dotenv" from the "devDependecies" to "dependencies".
+     * Change the "start" script in your server/package.json from "ts-node server.ts" to "node dist/server.js".
+     * Add a new script called "start:dev" with the value "ts-node server.ts". You can use this for your local development.
+   * The package.json file should look something like this:
+     * [`package.json`](Module13Assets/package.json) - Contains required dependencies and scripts
```

Comment:
Change from "Contains required dependencies and scripts" to "Should contain required dependencies and scripts, along with the dependencies you have added"

.ie, students should not by mistake copy-paste this file and then debug why their dependencies are missing

Reply:
Good point. Done